src/inmanta/agent/cache.py

```diff
@@ -76,21 +83,37 @@ def delete(self) -> None:
     def __del__(self) -> None:
         self.delete()
 
+    def refresh(self, now: float) -> None:
+        """
+        Refresh this cache item. Resets the expiry time for items marked
+        for eviction a certain time after their last access.
+
+        :parameter now: Baseline 'now' time to make sure expiry times are synced up
+            across cache items.
+        """
+        if self.refresh_after_access <= 0:
+            # Refreshing on access is disabled on the CacheItem.
+            return
+        self.expiry_time = min(now + self.refresh_after_access, self.after_creation_expiry_time)
```

Comment:
I see that there is a usage now, but it feels weird that it could impact the refresh time. I would definitively log a warning here

Reply:
While I agree that it's explicitly written:
```
If both are set, the shortest duration of the
    two will trigger the item to become stale
```
I did not understand that this would also apply for the refresh variant as well. Am I the only one?

Reply:
As I understood it, if both are set, whichever timer expires first makes the item stale.

Reply:
Never mind, after some discussions, I understood why it was like this